File: vinted-agent/audit_log.py
import logging
from supabase import AsyncClient

log = logging.getLogger(__name__)


async def audit_log(
    supabase: AsyncClient,
    actor_user_id: str,
    action: str,
    *,
    entity_type: str = None,
    entity_id: str = None,
    details: dict = None,
) -> None:
    try:
        await supabase.table("audit_logs").insert({
            "actor_type": "agent",
            "actor_user_id": actor_user_id,
            "action": action,
            "entity_type": entity_type,
            "entity_id": entity_id,
            "details": details or {},
        }).execute()
    except Exception as e:
        log.warning("audit_log failed: %s", e)

File: lib/utils/audit-log.ts
import { createServiceClient } from '@/lib/supabase/service';

type ActorType = 'user' | 'agent' | 'system';

export type AuditEntry = {
  actor_type: ActorType;
  actor_user_id?: string | null;
  action: string;
  entity_type?: string;
  entity_id?: string;
  details?: Record<string, unknown>;
};

export async function auditLog(entry: AuditEntry): Promise<void> {
  try {
    const svc = createServiceClient();
    await svc.from('audit_logs').insert(entry);
  } catch {
    // fire-and-forget — never block the main action
  }
}
